llama.cpp

```diff
@@ -1666,62 +1680,61 @@ static bool llama_kv_cache_init(
     cache.cells.clear();
     cache.cells.resize(n_ctx);
 
-    struct ggml_init_params params;
-    params.mem_size   = 2u*n_layer*ggml_tensor_overhead();
-    params.mem_buffer = NULL;
-    params.no_alloc   = true;
-
-    cache.ctx = ggml_init(params);
-
-    size_t vram_kv_cache = 0;
+    // count used buffer types
```

Comment:
It counts the number of times each type of buffer is used, because a different `ggml_context` needs to be allocated for each buffer type. The count is also used later to determine the size of the `ggml_context`.